### PR TITLE
[13.0][MIG] stock_product_pack: Migration to 13.0

### PR DESCRIPTION
**Pull request by joao-p-marques**
TT25572

ping @chienandalu @pedrobaeza

### CODE REVIEW

File: stock_product_pack/tests/test_stock_product_pack.py [hunk 1]
```diff
@@ -122,10 +81,13 @@ def test_compute_quantities_dict(self):
             }
         )
         picking.action_confirm()
-        self.assertEqual(self.pack_dc.virtual_available, 5)
-        self.assertEqual(self.pack_dc.qty_available, 0)
+        company = picking.company_id
+        pack = self.pack_dc.with_context(company_id=company.id)
```
Comment (chienandalu) [inline stock_product_pack/tests/test_stock_product_pack.py]: But the key to trigger the `depends_context` is `force_company`. Try it and let's see if the test succeds without the `invalidate_cache` trick

Comment (joao-p-marques) [reply stock_product_pack/tests/test_stock_product_pack.py]: I have tried passing `force_company=company.id` ans still no luck :disappointed:. Was this what you meant?

Comment (chienandalu) [reply stock_product_pack/tests/test_stock_product_pack.py]: Yes. Then assigning the context here is useless. Just `invalidate_cache` por this test

**Review by ernestotejeda — COMMENTED** [review]
Code review :+1:

**Review by ernestotejeda — APPROVED** [review]
Functional test :+1:

File: stock_product_pack/tests/test_stock_product_pack.py (hunk 1 above)
Comment (joao-p-marques) [reply stock_product_pack/tests/test_stock_product_pack.py]: Ok. Leaving this asis, then. Thank you anyway for the idea!

Comment (chienandalu) [reply stock_product_pack/tests/test_stock_product_pack.py]: What I mean is that this context is not very useful. So just add the line with `invalidate_cache`

Comment (joao-p-marques) [reply stock_product_pack/tests/test_stock_product_pack.py]: Ok. Changes made. Thank you.

**Review by rousseldenis — APPROVED** [review]
Code review

File: stock_product_pack/tests/test_stock_product_pack.py [hunk 2]
```diff
@@ -122,10 +81,12 @@ def test_compute_quantities_dict(self):
             }
         )
         picking.action_confirm()
-        self.assertEqual(self.pack_dc.virtual_available, 5)
-        self.assertEqual(self.pack_dc.qty_available, 0)
+        pack = self.pack_dc.with_context()
+        self.assertEqual(pack.virtual_available, 5)
+        self.assertEqual(pack.qty_available, 0)
```
Comment (chienandalu) [inline stock_product_pack/tests/test_stock_product_pack.py]: Just leave as it was
```suggestion
        self.assertEqual(self.pack_dc.virtual_available, 5)
        self.assertEqual(self.pack_dc.qty_available, 0)
```

File: stock_product_pack/tests/test_stock_product_pack.py [hunk 3]
```diff
@@ -122,10 +81,12 @@ def test_compute_quantities_dict(self):
...
         wizard_dict = picking.button_validate()
         wizard = self.env[wizard_dict["res_model"]].browse(wizard_dict["res_id"])
         wizard.process()
-        self.assertEqual(self.pack_dc.virtual_available, 5)
-        self.assertEqual(self.pack_dc.qty_available, 5)
+        self.product_obj.invalidate_cache()
+        self.assertEqual(pack.virtual_available, 5)
+        self.assertEqual(pack.qty_available, 5)
```
Comment (chienandalu) [inline stock_product_pack/tests/test_stock_product_pack.py]: Same here

```suggestion
        self.assertEqual(self.pack_dc.virtual_available, 5)
        self.assertEqual(self.pack_dc.qty_available, 5)
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
Just two last tiny changes:

File: stock_product_pack/tests/test_stock_product_pack.py (hunk 2 above)
Comment (joao-p-marques) [reply stock_product_pack/tests/test_stock_product_pack.py]: Of course :+1: . Making them now

Comment (joao-p-marques) [reply stock_product_pack/tests/test_stock_product_pack.py]: Changes made

**Comment by jjscarafia** [comment]
@ernestotejeda @pedrobaeza I believe this can be merged, isn't it?

**Review by chienandalu — APPROVED** [review]

**Comment by chienandalu** [comment]
:+1:

**Pull request merged**